x-pack/plugins/search_inference_endpoints/public/components/inference_flyout_wrapper_component.tsx

```diff
@@ -0,0 +1,151 @@
+import { extractErrorProperties } from '@kbn/ml-error-utils';
+import { TrainedModelConfigResponse } from '@kbn/ml-plugin/common/types/trained_models';
+import { SUPPORTED_PYTORCH_TASKS, TRAINED_MODEL_TYPE } from '@kbn/ml-trained-models-utils';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { INFERENCE_ENDPOINTS_QUERY_KEY } from '../../common/constants';
+import { useKibana } from '../hooks/use_kibana';
+import { docLinks } from '../../common/doc_links';
+
+interface InferenceFlyoutWrapperComponentProps {
+  isInferenceFlyoutVisible: boolean;
+  setIsInferenceFlyoutVisible: (isVisible: boolean) => void;
+}
+
+export const InferenceFlyoutWrapperComponent: React.FC<InferenceFlyoutWrapperComponentProps> = ({
+  isInferenceFlyoutVisible,
+  setIsInferenceFlyoutVisible,
+}) => {
+  const [inferenceAddError, setInferenceAddError] = useState<string | undefined>(undefined);
+  const [isCreateInferenceApiLoading, setIsCreateInferenceApiLoading] = useState(false);
+  const [availableTrainedModels, setAvailableTrainedModels] = useState<
+    TrainedModelConfigResponse[]
+  >([]);
+
+  const [inferenceEndpointError, setInferenceEndpointError] = useState<string | undefined>(
+    undefined
+  );
+
+  const queryClient = useQueryClient();
+
+  const {
+    services: { ml },
+  } = useKibana();
+
+  const createInferenceEndpointMutation = useMutation(
+    async ({
+      inferenceId,
+      taskType,
+      modelConfig,
+    }: {
+      inferenceId: string;
+      taskType: InferenceTaskType;
+      modelConfig: ModelConfig;
+    }) => {
+      await ml?.mlApi?.inferenceModels?.createInferenceEndpoint(inferenceId, taskType, modelConfig);
+    },
+    {
+      onSuccess: () => {
+        queryClient.invalidateQueries([INFERENCE_ENDPOINTS_QUERY_KEY]);
+      },
+    }
+  );
+
+  const onInferenceEndpointChange = async () => {};
+  const onSaveInferenceCallback = useCallback(
+    async (inferenceId: string, taskType: InferenceTaskType, modelConfig: ModelConfig) => {
+      setIsCreateInferenceApiLoading(true);
+      try {
+        await createInferenceEndpointMutation.mutateAsync({ inferenceId, taskType, modelConfig });
+        setIsInferenceFlyoutVisible(!isInferenceFlyoutVisible);
+      } catch (error) {
+        const errorObj = extractErrorProperties(error);
+        setInferenceAddError(errorObj.message);
+      } finally {
+        setIsCreateInferenceApiLoading(false);
+      }
+    },
+    [createInferenceEndpointMutation, isInferenceFlyoutVisible, setIsInferenceFlyoutVisible]
+  );
+
+  const onFlyoutClose = useCallback(() => {
+    setInferenceAddError(undefined);
+    setIsInferenceFlyoutVisible(!isInferenceFlyoutVisible);
+  }, [isInferenceFlyoutVisible, setIsInferenceFlyoutVisible]);
+
+  useEffect(() => {
+    const fetchAvailableTrainedModels = async () => {
+      setAvailableTrainedModels((await ml?.mlApi?.trainedModels?.getTrainedModels()) ?? []);
+    };
+    fetchAvailableTrainedModels();
```

Comment:
what happens if this throws an error? Have you tested this page with a user who does not have permission to ML or a basic license?

Reply:
I have added a try-catch around this section to handle exception. Besides this, we have a ticket to make the feature available to Enterprise licensed users only. I will address the licensing in a separate PR.